packages/yourloops/components/dashboard-widgets/medical-files/medical-report-list.util.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { type MedicalReport, type MedicalReportWithIndex } from '../../../lib/medical-files/models/medical-report.model'
+
+const getMedicalReportsOrderedByDate = (medicalReports: MedicalReport[]): Map<string, MedicalReportWithIndex[]> => {
+  return medicalReports.sort((firstMedicalReport, secondMedicalReport) => {
+    return firstMedicalReport.creationDate.localeCompare(secondMedicalReport.creationDate)
+  }).reduce((medicalReportsOrderedByDate: Map<string, MedicalReportWithIndex[]>, medicalReport) => {
+    const creationDate = medicalReport.creationDate.substring(0, 10)
+    const medicalReportToAdd: MedicalReportWithIndex = {
+      medicalReport
+    }
+    const isDateAlreadyIncluded = !!medicalReportsOrderedByDate.get(creationDate)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏Here the value could be fetched directly instead of checking only its presence, given that it's fetched again line 40 (I think it would work exactly the same given that `.get()` returns `undefined` if no value is found, but if you want to double check you can do a `!!` in the `if ()` statement)

```ts
const medicalReportsAtGivenDate = medicalReportsOrderedByDate.get(creationDate)
if (medicalReportsAtGivenDate) {
  medicalReportToAdd.index = medicalReportsAtGivenDate.length
  // ...
}
```